website/js/Splicing.js

```diff
@@ -0,0 +1,807 @@
+
+        return (
+            <g>
+                <text x={x + width / 2} y={height + 14} textAnchor="middle">{n}</text>
+
+                <g transform={isFlipped ? `translate(${x + width}) scale(-1,1)` : `translate(${x})`}>
+                    <defs>
+                        <clipPath id={clipMaskID}>
+                            <rect
+                                x={0} width={width} height={height} rx={exonBorderRadius} ry={exonBorderRadius}
+                                className={`segment exon`}
+                            />
+                        </clipPath>
+                    </defs>
+
+                    <rect x={0} className={`segment exon ${highlight && 'highlighted'}`} width={width} height={height} rx={exonBorderRadius} ry={exonBorderRadius} />
+
+                    <SegmentRegions n={n} variant={variant} donors={donors} acceptors={acceptors} CIDomains={CIDomains}
+                        selectedDomain={this.props.selectedDomain}
+                        txStart={txStart} txEnd={txEnd} zoomed={zoomed} isFlipped={isFlipped} isIntron={false}
+                        width={width} height={height} mask={clipMaskID} />
+
+                    {/* extra unfilled rect overlay, used to re-draw the outline on top of the regions */}
+                    <rect x={0} className="segment exon outline" width={width} height={height} rx={exonBorderRadius} ry={exonBorderRadius} />
+                </g>
+            </g>
+        );
+    }
+}
+
+class Intron extends React.Component {
+    render() {
+        const {
+            txStart, txEnd, x, height, width, variant, zoomed, highlight, isFlipped,
+            donors, acceptors, CIDomains
+        } = this.props;
+
+        return (
+            <g transform="translate(0, 10)">
+                <g transform={isFlipped ? `translate(${x + width}) scale(-1,1)` : `translate(${x})`}>
+                    <rect x={0} className={`segment intron ${highlight && 'highlighted'}`} width={width} height={height} />
+
+                    <SegmentRegions variant={variant} donors={donors} acceptors={acceptors} CIDomains={CIDomains}
+                        selectedDomain={this.props.selectedDomain}
+                        txStart={txStart} txEnd={txEnd} zoomed={zoomed} isFlipped={isFlipped} isIntron={true}
+                        width={width} height={height} />
+
+                    {/* extra unfilled rect overlay, used to re-draw the outline on top of the regions */}
+                    <rect x={0} className="segment intron outline" width={width} height={height} />
+                </g>
+            </g>
+        );
+    }
+}
+
+class Transcript extends React.Component {
+    render() {
+        const {variant, donors, acceptors,  CIDomains, segments, width, isFlipped} = this.props;
+
+
+        // ------------------------------------------------
+        // --- precalculate scale
+        // --- TODO: scales are slightly different between exonic / intronic variants. fix.
```

Comment:
followup

Reply:
I believe this ended up getting fixed when I switched to d3-scale, so I'll remove the comment.